core/capabilities/remote/target/request/caller_request.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+type callerRequest struct {
+	responseCh       chan commoncap.CapabilityResponse
+	createdAt        time.Time
+	responseIDCount  map[[32]byte]int
+	errorCount       map[string]int
+	responseReceived map[p2ptypes.PeerID]bool
+
+	requiredIdenticalResponses int
+
+	requestTimeout time.Duration
+
+	respSent bool
+	mux      sync.Mutex
+}
+
+func NewCallerRequest(ctx context.Context, lggr logger.Logger, req commoncap.CapabilityRequest, messageID string,
+	remoteCapabilityInfo commoncap.CapabilityInfo, localDonInfo capabilities.DON, dispatcher types.Dispatcher,
+	requestTimeout time.Duration) (*callerRequest, error) {
+	remoteCapabilityDonInfo := remoteCapabilityInfo.DON
+	if remoteCapabilityDonInfo == nil {
+		return nil, errors.New("remote capability info missing DON")
+	}
+
+	rawRequest, err := pb.MarshalCapabilityRequest(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal capability request: %w", err)
+	}
+
+	tc, err := transmission.ExtractTransmissionConfig(req.Config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract transmission config from request config: %w", err)
+	}
+
+	peerIDToTransmissionDelay, err := transmission.GetPeerIDToTransmissionDelay(remoteCapabilityDonInfo.Members, localDonInfo.Config.SharedSecret,
+		messageID, tc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get peer ID to transmission delay: %w", err)
+	}
+
+	responseReceived := make(map[p2ptypes.PeerID]bool)
+	for peerID, delay := range peerIDToTransmissionDelay {
+		responseReceived[peerID] = false
+		go func(peerID ragep2ptypes.PeerID, delay time.Duration) {
+			message := &types.MessageBody{
+				CapabilityId:    remoteCapabilityInfo.ID,
+				CapabilityDonId: remoteCapabilityDonInfo.ID,
+				CallerDonId:     localDonInfo.ID,
+				Method:          types.MethodExecute,
+				Payload:         rawRequest,
+				MessageId:       []byte(messageID),
+			}
+
+			select {
+			case <-ctx.Done():
+				return
+			case <-time.After(delay):
+				err := dispatcher.Send(peerID, message)
+				if err != nil {
+					lggr.Errorw("failed to send message", "peerID", peerID, "err", err)
+				}
+			}
+		}(peerID, delay)
+	}
+
+	return &callerRequest{
+		createdAt:                  time.Now(),
+		requestTimeout:             requestTimeout,
+		requiredIdenticalResponses: int(remoteCapabilityDonInfo.F + 1),
+		responseIDCount:            make(map[[32]byte]int),
+		errorCount:                 make(map[string]int),
+		responseReceived:           responseReceived,
+		responseCh:                 make(chan commoncap.CapabilityResponse, 1),
+	}, nil
+}
+
+func (c *callerRequest) ResponseChan() <-chan commoncap.CapabilityResponse {
+	return c.responseCh
+}
+
+func (c *callerRequest) Expired() bool {
+	return time.Since(c.createdAt) > c.requestTimeout
+}
+
+func (c *callerRequest) Cancel(reason string) {
```

Comment:
maybe better to accept an error than a string for type continuity, especially if the receiver of the capability response needs to handle specific kinds of errors